### ISSUE TITLE
Issue with thousand separator.

### ISSUE THREAD

**Issue by devsaurin** [paypal/PayPal-PHP-SDK#282]
As per Developer docs on Rest API page, https://developer.paypal.com/webapps/developer/docs/api/
it says:

----------------------------------------
**Amount should be non-negative number with a maximum of 2 decimal places separated by ‘.’, optional thousands separator ‘,’, limited to 7 digits before the decimal point**

-----------------------------------------

When I tried `setPrice()` and `SetTotal()` to set price with thousand separator and decimal points `number_format('1061.40', 2, '.', ',')` resulting number such as `1,062.40`. It gave me following error:

```PHP
Fatal error: Uncaught exception 'InvalidArgumentException' with message 'Total is not a valid numeric value' in 
/paypal/paypal/rest-api-sdk-php/lib/PayPal/Validation/NumericValidator.php:23 Stack trace: #0 
/paypal/paypal/rest-api-sdk-php/lib/PayPal/Api/Amount.php(57): PayPal\Validation\NumericValidator::validate('1,061.40', 'Total') #1
/paypal/CreatePaymentUsingPayPal.php(104): PayPal\Api\Amount->setTotal('1,061.40') #2
/paypalpay.php(92): require_once('/www/htdocs/w01...') #3 
{main} thrown in /paypal/paypal/rest-api-sdk-php/lib/PayPal/Validation/NumericValidator.php on line 23
```

I dig into the issue and found the statement in developer docs is not correct. If you set price with thousand separator, it generates an error.

**Technical Specs :**
---------------
`setPrice()` and `setTotal()` calls static function `validate()` from class `NumericValidator`, `NumericValidator::validate($total, "Total");`

Implementation of validation function is as follow:
    
```PHP
public static function validate($argument, $argumentName = null)
{
    if (trim($argument) != null && !is_numeric($argument)) {
        throw new \InvalidArgumentException("$argumentName is not a valid numeric value");
    }
    return true;
}
```
Where `is_numeric()` function in condition will always returns `false` for thousand separated amount like `1,061.40` and will raise an error of `"$argumentName is not a valid numeric value"`

**Solution :**
----------------------
In next step of setPrice()/setTotal(); `FormatConverter::formatToPrice()` function actually removes thousand separator by calling `self::formatToNumber()` function with `number_format($value, $decimals, '.', '');` usage. That clearly states that, internally PHP-SDK is using price without thousand separators. 

The aforementioned statement from PayPal Developer Docs https://developer.paypal.com/webapps/developer/docs/api/ & other related places  should be corrected. It should clearly be stated; not to use thousand separators and let developers handle price themselves.

Thanks.

**Comment by jaypatel512** [comment]
Hey @devsaurin ! Thank you so much for your detailed explanation. Really helpful. [Developer Docs](https://developer.paypal.com/webapps/developer/docs/api/) is a general API documentation, about PayPal REST API.

`price` and `total` field does take thousand comma separated value. However, to allow our SDKs to take in integer values, instead of strings with commas, we introduced the `NumericValidator` as well as `FormatConverter` in our PHP SDK. And as you can see, the thousand comma separator is optional, and SDK do not add them. However, if there is any reason that would make keeping thousand separator a better idea, I would definitely love to add that to the SDK.

So, as developer docs you pointed still holds true, for people not using our SDK, and trying to make direct calls using CURL, I would love to add more explanation/documentation to our SDKs, in our [wiki](https://github.com/paypal/PayPal-PHP-SDK/wiki) section to help SDK users.

**Comment by devsaurin** [comment]
Hello @jaypatel512 as I wrote, function to set price and total `setPrice()` and `setTotal()` when you assign thousand separated numerical string such as `1,048.60` it does shows error. That I already added to my previous message. That means the issue is not resolved that way. The `NumericaValidator` does not check for thousand separated numeric value. and `FormatConverter` actually removes thousand separator.
